src/Raven.Server/Rachis/RachisLogHistory.cs

```diff
@@ -489,5 +489,106 @@ public unsafe bool HasHistoryLog(ClusterOperationContext context, BlittableJsonR
                 return true;
             }
         }
+
+        public static long? TryGetLastRaftIndexForDeleteDatabaseCommandWithoutErrors(ServerStore serverStore, string databaseName, out bool hasExceptions)
+        {
+            var commandType = nameof(DeleteDatabaseCommand);
+            long? index = null;
+            hasExceptions = false;
+
+            using (serverStore.Engine.ContextPool.AllocateOperationContext(out ClusterOperationContext context))
+            using (context.OpenReadTransaction())
+            {
+                var table = context.Transaction.InnerTransaction.OpenTable(LogHistoryTable, LogHistorySlice);
+                var guidPrefix = $"{databaseName}/";
+                using (Slice.From(context.Allocator, guidPrefix, out var guidPrefixSlice))
+                {
+                    foreach (var (guidSlice, entryHolder) in table.SeekByPrimaryKeyPrefix(guidPrefixSlice, Slices.Empty, 0))
+                    {
+                        var entry = ReadHistoryLog(context, entryHolder);
+                        var curIndex = long.Parse(entry[nameof(LogHistoryColumn.Index)].ToString());
+                        var guid = entry[nameof(LogHistoryColumn.Guid)].ToString();
+                        var type = entry[nameof(LogHistoryColumn.Type)].ToString();
+                        var state = entry[nameof(LogHistoryColumn.State)].ToString();
```

Comment:
better use the `HistoryLogEntry` like in `CancelHistoryEntriesFrom` (make a method to read an entry to this class)